pkg/apis/kudo/v1beta1/operatorversion_types_helpers.go

```diff
@@ -6,24 +6,37 @@ import (
 
 	"k8s.io/apimachinery/pkg/types"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kudobuilder/kudo/pkg/util/kudo"
 )
 
 func OperatorInstanceName(operatorName string) string {
 	return fmt.Sprintf("%s-instance", operatorName)
 }
 
-func OperatorVersionName(operatorName, version string) string {
-	return fmt.Sprintf("%s-%s", operatorName, version)
+func OperatorVersionName(operatorName, ovVersion, appVersion string) string {
+	if appVersion == "" {
+		return fmt.Sprintf("%s-%s", operatorName, ovVersion)
+	}
+	return fmt.Sprintf("%s-%s-%s", operatorName, appVersion, ovVersion)
 }
 
 func (ov *OperatorVersion) FullyQualifiedName() string {
-	return fmt.Sprintf("%s-%s", ov.Name, ov.Spec.AppVersion)
+	return OperatorVersionName(ov.Spec.Operator.Name, ov.Spec.Version, ov.Spec.AppVersion)
 }
 
 func (ov *OperatorVersion) EqualOperatorVersion(other *OperatorVersion) bool {
 	return ov.FullyQualifiedName() == other.FullyQualifiedName()
 }
 
+func ListOperatorVersions(ns string, c client.Reader) (ovList *OperatorVersionList, err error) {
+	ovList = &OperatorVersionList{}
```

Comment:
naming nit: this is the only variable in this, four lines long, method. It hardly deserves such a long name, not to speak of the camel case. `l` would be more Go-like and exactly as clear to the reader.  This applies to a few other variables in this PR. I know, you're not a fan of Go-like variable names and longer variable names are sometimes meaningful 😉 . But here it seems out of place.

Reply:
Fair enough. I think this occasion stems from refactoring - I had this not separated before and `l` clashed with another var. I'll make a pass and look what's improvable.